src/app/TideMatchSet.cpp

```diff
@@ -869,6 +872,8 @@ void TideMatchSet::gatherTargetsAndDecoys(
     if (concat || !peptide.IsDecoy()) {
       if (targetsOut.size() < gatherSize) {
         targetsOut.push_back(i);
+      } else {
```

Comment:
Could you explain the goal of the 'else break'. I think you aim to break the for loop once targetsOut reaches gatherSize. However, wouldn't this mean that the number of gathered decoys would be incorrect? I'm thinking of the case where concat=F.

I'm thinking of the case where for a specific spectrum the top 2 matches are targets.

Reply:
Thanks for this catch. Indeed, it looks like a bug. I fixed it, I hope. Please take a look at it.